### PR TITLE
M2-4655: [Admin Panel] Validation error doesnt disappear when section/score/score condition name becomes unique

### PR DESCRIPTION
**Pull request by alex-i-e**
[M2-4655](https://mindlogger.atlassian.net/browse/M2-4655)

[M2-4655]: https://mindlogger.atlassian.net/browse/M2-4655?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/modules/Builder/features/ActivitySettings/ScoresAndReports/SectionScoreCommonFields/SectionScoreCommonFields.tsx [hunk 1]
```diff
@@ -87,6 +87,7 @@ export const SectionScoreCommonFields = ({
           control={control}
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivitySettings/ScoresAndReports/SectionScoreCommonFields/SectionScoreCommonFields.tsx:88]: Could you please modify the useEffect to include dependencies, or if necessary, disable the dependency check for useEffect?

File: src/modules/Builder/features/ActivitySettings/TitleComponent/TitleComponent.tsx [hunk 2]
```diff
@@ -24,6 +27,10 @@ export const TitleComponent = ({ title, name, open }: TitleComponentProps) => {
     : [];
   const isShowErrors = !open && errorMessages.length > 0;
 
+  useEffect(() => {
+    trigger(name);
+  }, [entities.length]);
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivitySettings/TitleComponent/TitleComponent.tsx]: Could you please modify the useEffect to include dependencies, or if necessary, disable the dependency check for useEffect?

File: src/modules/Builder/hooks/useCheckAndTriggerOnNameUniqueness.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { useCustomFormContext } from 'modules/Builder/hooks';
+
+export const useCheckAndTriggerOnNameUniqueness = ({
+  currentPath,
+  entitiesFieldPath,
+  checkIfShouldIncludeEntity = () => true,
+}: {
+  currentPath: string;
+  entitiesFieldPath: string;
+  checkIfShouldIncludeEntity?: (data: any) => boolean;
```
Comment (IhorNesterenko99) [inline src/modules/Builder/hooks/useCheckAndTriggerOnNameUniqueness.ts]: Please avoid using any.

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.test.tsx [hunk 4]
```diff
@@ -264,9 +267,13 @@ describe('SubscalesConfiguration', () => {
...
-    expect(associatedSubscaleElement).toHaveTextContent(
-      'Subscale: subscale_2 (Item: single_text_score)',
-    );
+    // wait for debounce callback
+    await waitFor(() => {
+      expect(associatedSubscaleElement).toHaveTextContent(
+        'Subscale: subscale_2 (Item: single_text_score)',
+      );
+    });
+
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.test.tsx:276]: Could you please remove unused variable which is in the ESLint warning?

File: src/modules/Builder/features/ActivitySettings/ScoresAndReports/SectionScoreCommonFields/SectionScoreCommonFields.tsx (hunk 1 above)
Comment (alex-i-e) [reply src/modules/Builder/features/ActivitySettings/ScoresAndReports/SectionScoreCommonFields/SectionScoreCommonFields.tsx:88]: good point! fixed

File: src/modules/Builder/hooks/useCheckAndTriggerOnNameUniqueness.ts (hunk 3 above)
Comment (alex-i-e) [reply src/modules/Builder/hooks/useCheckAndTriggerOnNameUniqueness.ts]: Added generic type for the hook

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**